---
title: Rps10, a molecular barcode with high sensitivity and specificity for oomycetes
output: html_document
---

```{r setup, include=FALSE}
source("style.R")
source("tools.R")
source("configuration.R")
```

**OomyceteDB** provides quality-controlled, aligned and annotated Oomycete *rps10* sequences.
*rps10* is a mitochondrial locus chosen as a molecular barcode suitable for metabarcoding or amplicon sequencing specific to the Oomycetes.
The curated database can be downloaded or queried using BLAST.
The following heat-tree shows the species currently represented in the database:

```{r}
library(metacoder)
seqs <- ape::read.FASTA(get_latest_release_fa())
obj <- extract_tax_data(names(seqs),
                        key = c(id = "info", name = "info", source = "info", tax_id = "info", taxonomy = "class"),
                        regex = "id=(.+)\\|name=(.+)\\|source=(.+)\\|tax_id=(.+)\\|taxonomy=(.+)",
                        class_sep = ";")
```

```{r cache=TRUE, fig.width=10, fig.height=8}
set.seed(2)
obj %>%
  filter_taxa(taxon_names == "Oomycetes", subtaxa = TRUE) %>%
  remove_redundant_names() %>%
  heat_tree(node_label = taxon_names,
            node_size = n_obs, 
            node_color = n_obs,
            node_size_range = c(0.01, 0.04),
            node_label_size_range = c(0.015, 0.03),
            edge_size_range =  c(0.01, 0.04) * .5,
            layout = "da", initial_layout = "re",
            node_color_axis_label = "Number of sequences")
```

Table: Species per genus currently included in **OomyceteDB**:

```{r}
library(dplyr)
library(knitr)
library(kableExtra)
taxa_in_table <- c("Phytophthora", "Pythium", "Phytopythium", "Achlya", "Albugo", "Peronospora", "Plasmopara", "Saprolegnia")
genus_table <- obj %>%
  filter_taxa(taxon_names %in% taxa_in_table, subtaxa = TRUE) %>%
  get_data_frame(c("n_obs", "taxon_names", "n_subtaxa")) %>%
  filter(taxon_names %in% taxa_in_table)
  

genus_table <- rbind(genus_table, list(paste0("**", sum(genus_table$n_obs), "**"),
                                       "**Total**",
                                       paste0("**", sum(genus_table$n_subtaxa), "**")))

genus_table %>%
  transmute("Genus" = taxon_names, "Number of species" = n_subtaxa, "Number of sequences" = n_obs) %>%
  kable() %>%
  kable_styling(bootstrap_options = "striped", full_width = F, position = "center") %>% 
  column_spec(1, italic = T) 
```


