---
title: Rps10, a molecular barcode with high sensitivity and specificity for oomycetes
output: html_document
---

```{r setup, include=FALSE}
source("style.R")
source("tools.R")
source("configuration.R")
library(metacoder)
library(ape)
library(dplyr)
library(knitr)
library(kableExtra)
library(purrr)
```

**OomyceteDB** provides quality-controlled, aligned and annotated Oomycete *rps10* sequences.
*rps10* is a mitochondrial locus chosen as a molecular barcode suitable for metabarcoding or amplicon sequencing specific to the Oomycetes.
The curated database can be downloaded or queried using BLAST (for any release).
We invite all members of the oomycete community to contribute sequences for new Oomycete species.
We will include any new taxon with a GenBank accession you share with us.

The following heat-tree shows the taxa currently represented in the database:


<!-- <iframe class = "shiny_frame" src="http://oomy.cgrb.oregonstate.edu:3838/grunwald/OomyceteDB_dev/contents" frameBorder="0">
</iframe> -->

```{r cache=FALSE, fig.width=12, fig.height=12}
rps10_db <- read_fasta(file.path("2020-10-22_release_1_rps10.fasta"))
rps10_data <- str_match(names(rps10_db), pattern = "name=(.+)\\|strain=(.+)\\|ncbi_acc=(.+)\\|ncbi_taxid=(.+)\\|oodb_id=(.+)\\|taxonomy=(.+)$")
colnames(rps10_data) <- c("header", "name", "strain", "ncbi_acc", "ncbi_taxid", "oodb_id", "taxonomy")
rps10_data <- as_tibble(rps10_data)

# Split genus into its own level
binomial <- map_chr(str_split(rps10_data$taxonomy, pattern = ';'), `[`, 7)
genus <- map_chr(str_split(binomial, pattern = '_'), `[`, 1)
rps10_data$taxonomy <- map_chr(seq_along(rps10_data$taxonomy), function(index) {
  sub(rps10_data$taxonomy[index], pattern = binomial[index], replacement = paste0(genus[index], ';', binomial[index]))
})

# Convert to taxmap
obj <- parse_tax_data(rps10_data, class_sep = ";", class_cols = 'taxonomy')

# Make plot
set.seed(2)
obj %>%
  filter_taxa(taxon_names == "Oomycetes", subtaxa = TRUE) %>%
  remove_redundant_names() %>%
  heat_tree(node_label = gsub(pattern = '_', replacement = ' ', taxon_names),
            node_size = n_obs, 
            node_color = n_obs,
            node_size_range = c(0.006, 0.04),
            node_label_size_range = c(0.0075, 0.03),
            layout = "da", initial_layout = "re",
            node_color_axis_label = "Number of sequences")


# set.seed(2)
# obj %>%
#   filter_taxa(taxon_names == "Oomycetes", subtaxa = TRUE) %>%
#   remove_redundant_names() %>%
#   heat_tree(node_label = ifelse(is_leaf, '', gsub(pattern = '_', replacement = ' ', taxon_names)),
#             node_size = n_obs, 
#             node_color = n_obs,
#             node_size_range = c(0.006, 0.04),
#             node_label_size_range = c(0.01, 0.03),
#             layout = "da", initial_layout = "re",
#             node_color_axis_label = "Number of sequences",
#             output_file = 'no_species_labels.pdf')


# set.seed(2)
# obj %>%
#   filter_taxa(taxon_names == "Oomycetes", subtaxa = TRUE) %>%
#   remove_redundant_names() %>%
#   heat_tree(node_label = ifelse(n_obs == 1 & is_leaf, '', gsub(pattern = '_', replacement = ' ', taxon_names)),
#             node_size = n_obs, 
#             node_color = n_obs,
#             node_size_range = c(0.006, 0.04),
#             node_label_size_range = c(0.008, 0.03),
#             layout = "da", initial_layout = "re",
#             node_color_axis_label = "Number of sequences",
#             output_file = 'greater_than_1_labels.pdf')

# set.seed(2)
# obj %>%
#   filter_taxa(taxon_names == "Oomycetes", subtaxa = TRUE) %>%
#   remove_redundant_names() %>%
#   heat_tree(node_label = ifelse(runif(length(taxon_names)) > .5 & is_leaf, '', gsub(pattern = '_', replacement = ' ', taxon_names)),
#             node_size = n_obs, 
#             node_color = n_obs,
#             node_size_range = c(0.006, 0.04),
#             node_label_size_range = c(0.008, 0.03),
#             layout = "da", initial_layout = "re",
#             node_color_axis_label = "Number of sequences",
#             output_file = 'random_subset_labels.pdf')
```

OomyceteDB currently includes the following Oomycete taxa:

```{r warning=FALSE}
taxa_in_table <- obj %>%
  filter_taxa(! taxon_names %in% c("unclassified_Phytophthora", "parasitica_species_group")) %>%
  filter_taxa(!is_leaf) %>% 
  filter_taxa(is_leaf) %>%
  taxon_names()
genus_table <- obj %>%
  filter_taxa(taxon_names %in% taxa_in_table, subtaxa = TRUE) %>%
  get_data_frame(c("n_obs", "taxon_names", "n_subtaxa")) %>%
  filter(taxon_names %in% taxa_in_table)
  
genus_table <- rbind(genus_table, list(paste0("**", sum(genus_table$n_obs), "**"),
                                       "**Total**",
                                       paste0("**", sum(genus_table$n_subtaxa), "**")))
genus_table %>%
  arrange(desc(as.numeric(n_subtaxa))) %>%
  transmute("Genus" = taxon_names, "Number of species" = n_subtaxa, "Number of sequences" = n_obs) %>%
  kable() %>%
  kable_styling(bootstrap_options = "striped", full_width = F, position = "center") %>% 
  column_spec(1, italic = T) 
```


