<!DOCTYPE html>
<html lang="en"><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <title>Phytophthora-ID | About</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="Niklaus J Grunwwald">

    <!-- Le styles -->
    <link href="Bootstrap_files/bootstrap.css" rel="stylesheet">
    <style>
      body {
        padding-top: 60px; /* 60px to make the container go all the way to the bottom of the topbar */
      }
    </style>
    <link href="Bootstrap_files/bootstrap-responsive.css" rel="stylesheet">

    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="../assets/js/html5shiv.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://twitter.github.com/bootstrap/assets/ico/apple-touch-icon-144-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="http://twitter.github.com/bootstrap/assets/ico/apple-touch-icon-114-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="http://twitter.github.com/bootstrap/assets/ico/apple-touch-icon-72-precomposed.png">
    <link rel="apple-touch-icon-precomposed" href="http://twitter.github.com/bootstrap/assets/ico/apple-touch-icon-57-precomposed.png">
    <link rel="shortcut icon" href="img/customIcon.png">
  </head>

  <body>

    <div class="navbar navbar-inverse navbar-fixed-top">
      <div class="navbar-inner">
        <div class="container">
          <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="brand" href="./index.html">Phytophthora-ID</a>
          <div class="nav-collapse collapse">
            <ul class="nav">
              <li class="./index.html"><a href="./index.html">Home</a></li>
              <li><a href="./seq-id.html">Sequence ID</a></li>
              <li><a href="./geno-id.html">Genotype ID</a></li>
              <li><a href="./about.html">About</a></li>            </ul>
          </div><!--/.nav-collapse -->
        </div>
      </div>
    </div>

    <div class="container">

      <h1>Protocols</h1>
      <h4>Internal transcribed spacer region (ITS) region</h4>
      <p>The nuclear ribosomal RNA (rRNA) genes (small subunit, large subunit and 5.8S) are organized in a cluster but separated by two internal transcribed spacer regions (ITS1 and ITS2):</p>
      <img src="img/its.gif" align="middle"/>
      <p>Typically, in eukaryotic genomes the rRNA genes are repeated in tandem arrays in the order of several hundred, or possibly thousands of copies (Hillis et al. 2006). The spacer regions ITS1 and ITS2 that are transcribed, but not translated, are more variable than the actual rRNA genes. The spacer regions diverge rapidly enough to be informative in distinguishing most (but not all) Phytophthora species (Cooke et al. 2000). Primers ITS4 and ITS6 are used for amplification of the ITS region in Phytophthora (White et al. 1990; Cooke and Duncan, 1997; Cooke et al. 2000).</p>

      <h4>Cox spacer region</h4>
      <p>The cytochrome c oxidase subunit 1 and 2 genes (cox1 and cox2, respectively) are present as a gene cluster (see highlighted area) on the mitochondrial chromosome in Phytophthora (as well as Pythium and many plant species)(Martin et al. 2007):</p>
      <img src="img/cox.png" align="middle"/>
      <p>Due to interspecific sequence variation the spacer region between these two genes can be used for development of species-specific markers (Martin et al. 2004, Tooley et al. 2006) as well as for sequence-based identification of species.</p>
      <h4>ITS-PCR</h4>
      <p>We generally do not isolate DNA, but instead obtain a tuft of mycelium from a culture. Scrape along the surface of a plate (1-2cm) toward the margins of a growing culture where sporangia are present on hyphae using a sterile, dry toothpick. Twirl toothpick into 100 µl PCR tube containing dH2O. Boil at 95.9°C for 5min. Proceed to PCR following conditions presented below:</p>
      <h5>Mix: </h5>
      <table class="table-striped">
          <thead>
            <tr>
              <th align="left">Component</th>
              <th align="left">Final concentration</th>
            </tr>
          </thead>
        <tbody>
          <tr>
              <td>DNA (Toothpick) </td>
              <td>
              <div>1-3 ng/µl </div>
              </td>
              </tr>
              <tr>
              <td>dNTP (2.0mM)</td>
              <td>
              <div>200 µM</div>
              </td>
              </tr>
              <tr>
              <td>ITS6 10uM stock</td>
              <td>
              <div>0.4 µM</div>
              </td>
              </tr>
              <tr>
              <td>ITS4 10uM stock</td>
              <td>
              <div>0.4 µM</div>
              </td>
              </tr>
              <tr>
              <td>10 x Taq Buffer with 15mM MgCl2</td>
              <td>
              <div>1 x</div>
              </td>
              </tr>
              <tr>
              <td>Taq 5u/µl</td>
              <td>
              <div>0.05 u/µl</div>
              </td>
              </tr>
              <tr>
              <td>dWater</td>
              <td>
              <div>NA</div>
              </td>
            </tr>
          </tbody>
      </table>

        <h5>Primers</h5>
        <table class="table-striped">
          <thead>
            <tr>
            <th align="left">Primer</th>
            <th align="left">Sequence</th>
          </thead>
          <tbody>
            <tr>
            <td>ITS4:</td>
            <td>5’-TCCTCCGCTTATTGATATGC-3’</td>
            </tr>
            <tr>
            <td>ITS6:</td>
            <td>5’-GAAGGTGAAGTCGTAACAAGG-3’</td>
            </tr>
          </tbody>
        </table>
        <p></p>
        <p><em>Phytophthora</em> spp. typically yield PCR products in the range of 862-941bp.</p>

        <h5>PCR program:</h5>
        <table class="table-striped">
          <thead>
            <tr>
              <th>Step</th>
              <th>Cycles</th>
              <th>Conditions</th>
              </tr>
          </thead>    
          <tbody>
            <tr>
                <td>1</td>
                <td>1</td>
                <td>94C for 3min;</td>
            </tr>
            <tr>
              <td>2<br>
                          3<br>
                          4 </td>
              <td>35 (repeat steps 2-4)</td>
              <td>
              <p>94C for 1min;<br>
                55C for 1min, <br>
                72C for 1min;</p>
              </td>
            </tr>
            <tr>
              <td>5</td>
              <td>1</td>
              <td>72C for 10min;</td>
            </tr>
            <tr>
              <td>-</td>
              <td>-</td>
              <td>Store at -20C until used</td>
            </tr>
          </tbody>
        </table>

      <h4>Cox spacer PCR</h4>
      <p>The primer annealing sites are in the flanking gene sequences and are specific for amplification of Phytophthora spp. (they will not amplify the related genus Pythium or plant DNA when the described amplification conditions are followed):</p>
      <img src="img/cox-cartoon.png" align="middle">
      <p>Several considerations when using this marker system:</p>
      <ul>
        <li>The primers FMPhy-8b and FMPhy-10b should be used if amplification is done with DNA purified from infected plant tissue</li>
        <ul>
          <li>If amplification is done using DNA purified from single cultures the alternative primer listed below can be used for amplification (they will amplify sequences from Pythium but not plants) or for sequencing of templates.</li>
          <li>The amplicons are in the range of approximately 460 bp in size (approximately 230 bp of spacer sequence), but can vary among species.</li>
          <li>There are several plant species that will give nonspecific background amplification if the proper amplification stringency is not followed. Techniques for validating amplification conditions are outlined below and should be followed if amplification is done from DNA extracted from infected plant tissue.</li>
        </ul>
        <li>If DNA purified from single cultures is used the amplification stringency requirements are not as important
        </li>
        <li>In some cases the pathogen target DNA concentration is low relative to host plant DNA, so a second round of amplification using the same primers may be necessary to visualize the Phytophthora genus specific amplicon.
        </li>
        <li>A low level of intraspecific variation in sequences has been identified for some species, so it is possible to have several different sequence entries in the database for the same species.
        </li>
        <li>When multiple Phytophthora spp. are present in an infected lesion the amplicons will need to be cloned prior to sequencing.
        </li>
        <li>RFLP analysis of this amplicon can be used to differentiate many, but not all species. This data is currently being pulled together and will be available at a later date on the Phytophthora database website (www.phytophthoradb.org).
        </li>
      </ul>

      <h5>First Amplification Master Mix: </h5>
      
      <table class="table-striped">
        <tbody>
        <tr>
        <td width="71%"><strong>Component</strong></td>
        <td width="29%">
        <div align="right"><strong>Final                concentration </strong></div>
        </td>
        </tr>
        <tr>
        <td>dNTP</td>
        <td>
        <div align="right">0.5                µl</div>
        </td>
        </tr>
        <tr>
        <td>FMPhy-8b (12.5 µM)</td>
        <td>
        <div align="right">2.0                µl </div>
        </td>
        </tr>
        <tr>
        <td> FMPhy-10b (12.5 µM)</td>
        <td>
        <div align="right">2.0                µl</div>
        </td>
        </tr>
        <tr>
        <td>MgCl2 (3 mM)</td>
        <td>
        <div align="right">3.0                µl</div>
        </td>
        </tr>
        <tr>
        <td>AmpliTaq (N8080172)</td>
        <td>
        <div align="right">0.2                µl </div>
        </td>
        </tr>
        <tr>
        <td>Glycerol (50%; Roche cat.              # 100 647)</td>
        <td>
        <div align="right">1.0                µl </div>
        </td>
        </tr>
        <tr>
        <td>Buffer</td>
        <td>
        <div align="right">2.5                µl </div>
        </td>
        </tr>
        <tr>
        <td>dWater (Sigma cat. # W4502)</td>
        <td>to final volume of 25              µl</td>
        </tr>
        </tbody>
       </table>
      <h5>Primers:</h5>
      <table class="table-striped">
        <tbody>
        <tr>
        <td width="19%"><strong>Primer</strong></td>
        <td width="81%"><strong>Sequence</strong></td>
        </tr>
        <tr>
        <td>FMPhy-8b:</td>
        <td>5’-AAAAGAGAAGGTGTTTTTTATGGA-3’</td>
        </tr>
        <tr>
        <td>FMPhy-10b</td>
        <td>5’-GCAAAAGCACTAAAAATTAAATATAA-3’</td>
        </tr>
        </tbody>
      </table>

      <h5>Program:</h5>
      <table class="table-striped">
        <tbody>
        <tr>
        <td width="12%"><strong>Step</strong></td>
        <td width="34%"><strong>Cycles</strong></td>
        <td width="54%"><strong>Conditions</strong></td>
        </tr>
        <tr>
        <td width="12%">1</td>
        <td width="34%">1</td>
        <td width="54%">95C for 3min;</td>
        </tr>
        <tr>
        <td>2<br>
                    3<br>
                    4 </td>
        <td>40 (repeat steps 2-4)</td>
        <td>
        <p>95C for 1min;<br>
                    65.5C for 1min, <br>
                    72C for 1min;</p>
        </td>
        </tr>
        <tr>
        <td>5</td>
        <td>1</td>
        <td>72C for 5min;</td>
        </tr>
        <tr>
        <td>-</td>
        <td>-</td>
        <td>Store at -20C until used</td>
        </tr>
        </tbody>
      </table>
      <h5>Alternative primers:</h5>
      <ul>
      <li> FMPh-8 (dAAGGTGTTTTTTATGGACAATGTA)– first 17 bp is the 3’ end of FMPhy-8b</li>
      <li> Amplification with this primer and FMPhy-10b is more robust than FMPhy-8b under the amplification conditions used and will generate greater 
      amounts of amplified product. One possible reason for this is
       the higher stringency requirements for FMPhy-8b + FMPhy-10b 
      reduces the amount of product amplification.</li>
      <li> This primer pair should be used with caution where the presence of a genus-specific band alone is used for diagnostic purposes.
      <ul>
      <li> While they do not amplify the plant species that have been tested, they will amplify <em>Pythium</em> spp.</li>
      <li>Since <em>Pythium</em> spp. are found predominantly to cause diseases of below ground plant parts, for detection of foliar diseases 
      caused by <em>Phytophthora</em> spp. they should work well.</li>
      </ul>
      </li>
      <li> Good amplification at 3 or 4 mM Mg, weaker at 2 mM and none at 1 mM</li>
      <li>Amplification done using the same procedure as FMPhy-8b + 10B except that 3.5 µl of 50% glycerol was added. </li>
      <li>This primer can also be used as a nested sequencing primer in the event the amplification with FMPhy-8b and FMPhy-10b did not yield a 
      single amplicon.</li>
      </ul>
      <h5>Notes on amplification:</h5>
      <ul>
      <li> Start the amplification program on the thermal cycler and put on “pause” once the initial denaturation temperature is reached.</li>
      <li>The tubes are placed in the wells, the lid closed and the program restarted.</li>
      <li>After amplification, 5 µl of the reaction mix is run on a 1.5% gel (a 3% NuSieve 3:1 gel is used            if sharp bands are desired) to visualize the 
      bands.</li>
      <li> When amplification is 
      done            from DNA extracted from infected plant tissue sometimes a
       single round            of amplification will not generate a <em>Phytophthora</em>
       genus-specific            amplicon due to the low pathogen DNA 
      concentration relative to the plant            DNA in the sample. So 
      even if a <em>Phytophthora</em> specific band            is not seen, 
      continue with the second round of amplification (while            this 
      is an issue with forest samples, it may not with samples form ornamental
                  plants). </li>
      <li>The presence of glycerol  
                enhances the specificity of the amplification, without it a 
      band may            be amplified by some, but not all, plant species.</li>
      </ul>

      <h5>Second Round Amplification</h5>
      <p>Samples from the first round amplification are diluted 1:100. </p>
      
      <h5>Second Amplification Master Mix:</h5>

      <table border="1" width="80%">
        <tbody>
        <tr>
        <td>Component</td>
        <td>
        <div align="right"><strong>Final concentration </strong></div>
        </td>
        </tr>
        <tr>
        <td>Buffer</td>
        <td>
        <div align="right">2.5 µl</div>
        </td>
        </tr>
        <tr>
        <td>MgCl2 (3 mM)</td>
        <td>
        <div align="right">3.0 µl</div>
        </td>
        </tr>
        <tr>
        <td>dNTP mix</td>
        <td>
        <div align="right"> 0.5 µl </div>
        </td>
        </tr>
        <tr>
        <td>FMPhy-8b (12.5 µM)</td>
        <td>
        <div align="right">0.5 µl </div>
        </td>
        </tr>
        <tr>
        <td>FMPhy-10b (12.5 µM)</td>
        <td>
        <div align="right">0.5 µl </div>
        </td>
        </tr>
        <tr>
        <td>AmpliTaq</td>
        <td>
        <div align="right">0.2 µl </div>
        </td>
        </tr>
        <tr>
        <td> Diluted DNA from first round of amplification</td>
        <td>
        <div align="right">1.0 µl</div>
        </td>
        </tr>
        <tr>
        <td>dWater (Sigma cat. # W4502)</td>
        <td>
        <div align="right">to final volume of 25 µl</div>
        </td>
        </tr>
        </tbody>
      </table>
 
      <h5>Program for second amplification:</h5>
      <table border="1" width="68%">
        <tbody>
        <tr>
        <td width="12%"><strong>Step</strong></td>
        <td width="34%"><strong>Cycles</strong></td>
        <td width="54%"><strong>Conditions</strong></td>
        </tr>
        <tr>
        <td width="12%">1</td>
        <td width="34%">1</td>
        <td width="54%">95C for 3min;</td>
        </tr>
        <tr>
        <td>2<br>
                    3<br>
                    </td>
        <td>35 (repeat steps 2-3)</td>
        <td>
        <p>95C for 30sec;<br>
                    64C for 30sec, <br>
                    </p>
        </td>
        </tr>
        <tr>
        <td>4</td>
        <td>1</td>
        <td>72C for 5min;</td>
        </tr>
        <tr>
        <td>-</td>
        <td>-</td>
        <td>Store at -20C until used</td>
        </tr>
        </tbody>
      </table>
      <h5>General Notes:</h5>
      <ul>
      <li> The above procedures have been validated using and ABI 9700 thermalcycler and the indicated sources of reagents and Taq polymerase. In our experience changing any of these may affect the results 
      obtained and require revalidation of the amplification        
      conditions (especially if amplification is done using DNA extracted form infected plant tissue). </li>
      <li>One of the first things that will need to be tested when this procedure is used on a different PCR machine is to evaluate the effect of annealing 
      temperature on the specificity of amplification as 
      differences have been encountered when using different 
      thermalcyclers (this is not as important if DNA from purified
       cultures is used in the amplification).
      <ul>
      <li> The annealing temperatures we used with our thermalcycler (ABI 9700) 
      for the first and second round amplification was 65.5 C 
      and 64 C, respectively. However, when we used a MJ unit 
      the temperatures had to be reduced by 1 C for the marker 
      system to work. We believe this is due to differences in 
      block calibration between machines (possibly differences in ramp        
      rates as well) and highlights the importance of validating your 
      machine before getting started with using the diagnostic 
      assay. Several suggestions: 
      <ul>
      <li>use purified Phytophthora DNA and healthy plant DNA from the hosts you will be evaluating to check the annealing temps for 
      specificity. If the annealing temp is too low for the
      first round amplification some plant DNA may give a 
      band the same size as the Phytophthora genus specific
       amplicon. </li>
      <li>After the optimum annealing temperatures have been determined the plant and Phytophthora DNA are combined in the master mix and 
      12 tubes are placed in a uniform pattern across the 
      thermalcycler block to test for block uniformity. 
      Both the first and second amplification cycles are 
      tested this way and the separate tubes are compared for                 
      intensity of amplification and lack of background bands. To          
      ensure our thermalcycler is functioning properly we repeat 
      this test on a regular basis.</li>
      <li> If the annealing         
      temp for the first round amplification with the Phytophthora 
      genus specific primer pair is to high there will be a
      low level of genus specific amplicon generated, 
      which will reduce the sensitivity of detection.</li>
      <li> Under the conditions indicated above Phy-8b + 10b primer pair amplifies all of the <em>Phytophthora</em> species with the exception of low amplification of P. lateralis and P. sojae. 
      <ul>
      <li>Both these species are amplified by the earlier version of this primer 
      pair, Phy-8 + 10.</li>
      <li> While this primer pair will amplify all Phytophthora spp. and none of the plant material tested, it also amplifies Pythium spp. as well. For this reason it cannot be used for assaying any plant part that comes into 
      contact with the soil.</li>
      </ul>
      </li>
      </ul>
      </li>
      </ul>
      </li>
      </ul>
      <h5>Trouble shooting:</h5>
      <ul>
      <li>Bands in the 450-500 bp range when only plant DNA is in the amplification mixture – the stringency needs to be increased. Make sure the glycerol is 
      included in the master mix, increase the annealing temperature by 1 C,</li>
      <li> Faint <em>Phytophthora</em> genus-specific band – the annealing temperature may be too 
      high for the first round amplification with the 
      genus-specific primer pair. Run samples with <em>Phytophthora</em>
       and plant DNA individually in separate tubes to optimize the
       annealing temperatures to prevent background  amplification 
      of plant DNA by the genus-specific primers. With the two            
      round nested amplification procedure as little as 2 fg total <em>Phytophthora</em> DNA should be detected.</li>
      </ul>
      <p>Additional details on using this region for diagnostics can be found at <a href="http://www.ars.usda.gov/Research/docs.htm?docid=8733" target="_blank">http://www.ars.usda.gov/Research/docs.htm?docid=8733</a>.</p>
 
      <h4>Sequencing</h4>
      <p>Confirm presence of one 
      amplicon by agarose gel electrophoresis. Follow instructions 
      from your sequencing center to obtain ITS sequence. Sequence 
      twice in forward and reverse directions.</p>
      <p>Once sequence is returned 
      to you, usually in form of an &lt;abi&gt; file, check sequence 
      for quality (use a program such as <a href="http://www.mbio.ncsu.edu/BioEdit/BioEdit.html">Bioedit</a>). ITS or <em>Cox </em>spacer sequences should be sequenced in both directions. If sequences are clean proceed to either ITS-BLAST or <em>Cox</em>-BLAST (links to your left) and submit sequence in FASTA format for species identification.</p>
      <p>Occasionally you will observe polymorphic sites or sequences that abruptly end probably due to presence of and indel or because a 
      hybridization event occurred. In this case additional work is 
      needed to identify a species. A suggested next step included cloning    
      and sequences to determine haplotypes and/or sequencing of 
      additional nuclear or mitochondrial genes and BLAST search in <a href="http://www.phytophthoradb.org/" target="_blank">PhytophthoraDB</a>.</p>

      <h4>Citation</h4>
      <p>If you use this Phytophthora-ID for publishable research please cite the following reference:</p>

      <p>Grünwald,  N. J., Martin, F. N., Larsen, M. M., Sullivan, Press, C. M., Coffey, M.  D., Hansen, E. M., and Parke, J. L. 2011. Phytophthora-ID.org: A sequence-based Phytophthora identification tool. Plant Disease 95: 337-342.</span></p>

      <h4>References</h4>
      <p> Cooke, D. E. L., and Duncan, J. M. 1997. Phylogenetic analysis of <em>Phytophthora</em> species based on ITS1 and ITS2 sequences of the ribosomal RNA gene repeat. Mycol. Res. 101:667-677 </p>
      <p> Cooke, D. E. L., Drenth, A., Duncan, J. M., Wagels, G., and Brasier, C. M. 2000. A molecular phylogeny of <em>Phytophthora</em> and related oomycetes. Fungal Genet. Biol. 30:17-32.</p>
      <p>Grünwald,  N. J., Martin, F. N., Larsen, M. M., Sullivan, Press, C. M., Coffey, M.  D., Hansen, E. M., and Parke, J. L. 2011. Phytophthora-ID.org: A sequence-based Phytophthora identification tool. Plant Disease 95: 337-342.</p>
      <p>Hillis, D. M., Moritz, C., and Mable, B. K. 1996. Molecular Systematics. Sinauer Associate, Inc., Sunderland, MA. </p>
      <p>Martin, F.N, Tooley, P.W. and Blomquist, C. 2004. Molecular detection of <em>Phytophthora ramorum</em>, the causal agent of sudden oak death in California, and two additional species commonly recovered from diseased plant 
      material. Phytopathology 94:621-631.</p>
      <p>Martin F.N., Bensasson D., Tyler B.M., Boore J.L. 2007. Mitochondrial genome sequences and comparative genomics of <em>Phytophthora ramorum </em>and <em>P. sojae</em>. Current Genetics 51:285-296.</p>
      <p>Tooley, P.W., Martin, F.N., Carras, M.M. and Frederick, R.D. 2006. Real-time fluorescent PCR detection of <em>Phytophthora</em> <em>ramorum</em> and <em>Phytophthora</em> <em>pseudosyringae</em> using mitochondrial gene regions. Phytopathology 96: 336-345</p>
      <p>White, T. J., Bruns, T., Lee, S., and Taylor, J. 1990. Amplification and direct sequencing of fungal ribosomal RNA genes for phylogenetics. 
      Pages 315-322 in: PCR protocols: a guide to methods and 
      applications, M. A. Innis, D. H. Gelfand, J. J. Sninsky and T. 
      J. White, eds. Academic Press, San Diego.</p>
    </div>

 
      <hr>

      <footer>
        <p><small><a href="http://grunwaldlab.cgrb.oregonstate.edu/" target="_blank"> <strong>Grünwald lab</strong></a>
        <br><strong>Horticultural Crops Research Laboratory, USDA ARS</strong>                
        <br><strong>Department of Botany and Plant Pathology, Oregon State University</strong>
        <br>3420 NW Orchard Ave., Corvallis, OR 97330 </small>
      </footer>


    </div> <!-- /container -->

    <!-- Le javascript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="Bootstrap_files/jquery.js"></script>
    <script src="Bootstrap_files/bootstrap-transition.js"></script>
    <script src="Bootstrap_files/bootstrap-alert.js"></script>
    <script src="Bootstrap_files/bootstrap-modal.js"></script>
    <script src="Bootstrap_files/bootstrap-dropdown.js"></script>
    <script src="Bootstrap_files/bootstrap-scrollspy.js"></script>
    <script src="Bootstrap_files/bootstrap-tab.js"></script>
    <script src="Bootstrap_files/bootstrap-tooltip.js"></script>
    <script src="Bootstrap_files/bootstrap-popover.js"></script>
    <script src="Bootstrap_files/bootstrap-button.js"></script>
    <script src="Bootstrap_files/bootstrap-collapse.js"></script>
    <script src="Bootstrap_files/bootstrap-carousel.js"></script>
    <script src="Bootstrap_files/bootstrap-typeahead.js"></script>

  

</body></html>