---
title: "Rps10 database sequence PCR protocol"
output:
  html_document: default
  pdf_document:
    latex_engine: xelatex
---

```{r setup, include=FALSE}
source("style.R")
```

*Updated August 2019*


**The purpose of this protocol is to amplify the oomycete rps10 locus for use in the rps10 barcode database. If you want to use the Rps10 locus for amplicon metagenomics, use the metabarcoding protocol instead.**
The rps10 database amplicon is longer than the rps10 locus-specific amplicon and conveniently includes the rps10 locus-specific primer sequences.
The amplicon produced is usually around 554bp long.


## Ordering rps10 database primers

Both the rps10 forward and reverse database primer binding sites have two SNPs represented with IUPAC codes (Y=C,T and R=A, G). 
We ordered the Rps10 database oligos from Life technologies. 

```{r}
custom_kable(tibble(
  "Direction" = c("forward", "reverse"),
  "Primer name" = c("rps10_DB-FOR", "rps10_DB-REV"),
  "Primer sequence (5' -> 3')" = c("GTTGGTTAGAGYARAAGACT", "RTAYACTCTAACCAACTGAGT")
), escape = opts_knit$get("rmarkdown.pandoc.to") == 'latex')
```


## Rps10 database PCR mixture

```{r}
data <- read_tsv("tables/database_pcr_recipe.tsv")
data[c(2, 7), 1] <- paste0(unlist(data[c(2, 7), 1]), footnote_marker_number(1))
data[3, 1] <- paste0(unlist(data[3, 1]), footnote_marker_number(2))
data[4, 1] <- paste0(unlist(data[4, 1]), footnote_marker_number(3))
custom_kable(data, escape = FALSE) %>%
   footnote(number = c("GenScript Taq and 10x buffer from GenScript (Cat. No. E00007)",
                       "25mM MgCl2 from USB (Cat. No 71167)",
                       "100mM dNTP from Invitrogen (Cat. No 10297-018)"))
```


## Rps10 database thermocycler conditions

```{r}
custom_kable(read_tsv("tables/database_pcr_thermocycler_profile.tsv")) 
```
